src/main/java/no/kriops/drystreak/DryStreakCalculation.java

```diff
@@ -22,26 +20,52 @@ public class DryStreakCalculation implements ActionListener {
     @Override
     public void actionPerformed(ActionEvent e) {
         final String explanation = "At this number of kills, you are less likely to have received zero drops than not.";
-        final String input = inputField.getText();
-        Try
-                .of(() -> 1.0 / Integer.parseInt(input))
-                .orElse(Try.of(() -> Double.parseDouble(input)))
-                .orElse(Try.of(() -> Fraction.getFraction(input).doubleValue()))
-                .toEither("Unable to parse input.")
-                .filterOrElse(p -> 0 < p && p <= 1, integer -> "Drop rate must be greater than zero.")
-                .flatMap(p -> Try.of(() -> new GeometricDistribution(null, p))
-                        .mapTry(dist -> dist.inverseCumulativeProbability(0.5))
-                        .toEither("Error performing calculation."))
-                .map(result -> String.format("<p>Expected dry streak: %s</p><br><p>%s</p>", result, explanation))
-                .map(this::asHtml)
-                .peek(outputField::setText)
-                .mapLeft(this::asHtml)
-                .peekLeft(outputField::setText)
-        ;
+        String message = Optional.ofNullable(inputField.getText())
+            .map(input -> {
+                try {
+                    return 1.0 / Integer.parseInt(input);
+                } catch (NumberFormatException ignored) {
+                }
+
+                try {
+                    return Double.parseDouble(input);
+                } catch (NumberFormatException ignored) {
+                }
+
+                try {
+                    return Fraction.getFraction(input).doubleValue();
+                } catch (NumberFormatException ignored) {
+                }
+
+                return null;
+            })
+            .map(p -> {
+                if (0 < p && p < 1) {
+                    try {
+                        int result = geometricMedian(p);
+                        return String.format("<p>Expected dry streak: %s</p><br><p>%s</p>", result, explanation);
+                    } catch (Exception ignored) {
+                        return "Error performing calculation.";
+                    }
+                } else {
+                    return "Drop rate must be greater than zero and less than one.";
+                }
+            })
+            .orElse("Unable to parse input.");
+
+        outputField.setText(asHtml(message));
     }
 
-    private String asHtml(String input) {
+    private static String asHtml(String input) {
         return String.format("<html>%s</html>", input);
     }
 
+    private static int geometricMedian(double p) {
+        return Math.max(0, (int) Math.ceil(-1 / log2(1 - p)) - 1);
```

Comment:
Since `inverseCumulativeProbability(0.5)` is the only desired statistic, we just use the simpler [formula](https://wikimedia.org/api/rest_v1/media/math/render/svg/33f1a3cb0f6046156cedb3f498cfdeb076c71793) for the median of the geometric distribution